composer.json

```
@@ -0,0 +1,47 @@
+{
+    "name": "m6web/rate-limit-bundle",
+    "type": "symfony-bundle",
+    "authors": [
+        {
+            "name": "M6Web",
+            "email": "opensource@bedrockstreaming.com",
+            "homepage": "https://tech.bedrockstreaming.com/"
+        }
+    ],
+    "config": {
+        "bin-dir": "bin",
+        "vendor-dir": "vendor",
+        "sort-packages": true
+    },
+    "require": {
+        "php": "7.4.*",
+        "ext-json": "^7.4",
```

```suggestion
        "ext-json": "*",
```

I think that we don't really care about the ext-json version used (if we just need `json_decode`/`json_encode`)

---

>Benoit Viguier 
@lesDevsPhp Dans le composer.json, c’est très bien de mettre les dépendances aux extensions (genre "ext-json": "*"), ça permet de vérifier que l’extension est bien là avant de déployer en prod! (lors du composer install). Mais il faudrait qu’on prenne l’habitude de mettre une contrainte sur la version (au moins la version => ^5.0 par exemple). Sinon, lorsqu’on met à jour l’image Php, que par hasard des extensions sont mises à jour, que la sonde “status” fonctionne (car on teste jamais toutes les fonctions d’une extension dans une route status), et bien ça peut faire :boum: en prod :perplexe:

---

I don't think that this comment works with opensourced libraries

This bundle is already way too restrictive IMO, so we should at least leave users use the `ext-json` they want (they'll install a `ext-json` version which works with php7.4 and everyone will be happy)

---

I'll set this dependency with `*` requirements.
We don't use features added with php 7, so we don't care about the version.

Nikita Popov advice to set this requirement to `*` in this comment : https://github.com/lcobucci/content-negotiation-middleware/issues/31#issuecomment-547817869

@b-viguier What do you think about that?

---

Ok, you are extracting from slack archives a comment that I made 1 yr ago… you're so tough with me 😅 

As far as I remember, this comment was done after I encountered a similar issue with curl extension during an upgrade of a Docker image, where some constants have been deprecated. In this case, a more restrictive constraint would have save our production.
I'm not aware as Nikita Popov about versioning strategies for built-in extensions, so is advice is relevant, especially for an OSS library. `ext-json` seems pretty stable with a well defined scope, so I'm not horrified if you choose `*` 😉 
My general advice is that it's easier to start by _strict_ rules and relax them later, rather to impose stricter rules later 🤷 

Have fun 😅

---

I forgot this thread will be public, i will probably erase `slack` link and clean message.